rllib/agents/trainer.py

```diff
@@ -1055,12 +1055,14 @@ def __getstate__(self):
         return state
 
     def __setstate__(self, state):
-        if "worker" in state:
+        if "worker" in state and hasattr(self, "workers") and \
+                isinstance(self.workers, WorkerSet):
```

Comment:
Hmm do you know why PPO doesn't have the workers attribute? Shouldn't it always be present for that?

Reply:
Very good question :) I have no idea at this point. I found lots of these checks, also for other props and therefore did not wonder much about it. But yeah, it seems like bad practice to not reliably have these in any instance of Trainer.

Reply:
I can dig further. May be a deeper issue that we should fix.

Reply:
Sounds good. Maybe we are calling restore before the constructor is finished or something weird like that.

Reply:
Any updates on this?